### PR TITLE
clean up

### LINKED ISSUE #126: SensitivityOp interface
The output of the SensitivityOp should be defined via type hinting: https://github.com/PTB-MR/mrpro/blob/770376be712b94bf44e520767fed89cfeb48321c/src/mrpro/operators/_SensitivityOp.py#L26C23-L26C31

`forward(self, img_data: torch.Tensor) -> torch.Tensor`: 
and without `torch.Tensor` in `Returns`

Same for adjoint.

### PR DESCRIPTION
**Pull request by ckolbPTB**
This fixes #126 and hopefully makes the rest of the code a bit more consistent.

### CODE REVIEW

File: src/mrpro/operators/_FourierOp.py [hunk 1]
```diff
@@ -173,7 +179,7 @@ def get_target_pattern(fft_dims: tuple[int, ...], nufft_dims: tuple[int, ...], i
             elif dim in ignore_dims:
                 ignore_dims_str += ' ' + dim_str
 
-        target_pattern = f'(other{fft_dims_str}{ignore_dims_str}) coil{nufft_dims_str}'
+        target_pattern = f'(other{fft_dims_str}{ignore_dims_str}) coils{nufft_dims_str}'
```
Comment (fzimmermann89) [inline src/mrpro/operators/_FourierOp.py:182]: If you change coil -> coils here, you also have to change it when this pattern is used.
Also, maybe  make other capture multiple dimensions already via ... ?

File: src/mrpro/operators/_FourierOp.py [hunk 2]
```diff
@@ -262,7 +268,7 @@ def adjoint(self, y: torch.Tensor) -> torch.Tensor:
 
         # move dim where FFT was already performed such nuFFT can be performed
         if len(self._nufft_dims) != 0:
-            init_pattern = 'other coil dim2 dim1 dim0'
+            init_pattern = 'other coils dim2 dim1 dim0'
```
Comment (fzimmermann89) [inline src/mrpro/operators/_FourierOp.py:271]: same as above

File: src/mrpro/operators/_FourierOp.py (hunk 1 above)
Comment (fzimmermann89) [reply src/mrpro/operators/_FourierOp.py:182]: example would be line 230

**Review by fzimmermann89 — CHANGES_REQUESTED** [review]

**Comment by fzimmermann89** [comment]
Re the other dimension:

I would prefer if we allow in our code for this dimension to actually be more than 1 dimensnion.
So we dont split the KData into dynamics, ... by default, but our code works if it is split.

This would, as a start, mean replacing 'other' by '...' in all einops pattern, and counting dimensions from the back for swapaxes etc.

What do you think, Christoph?

**Review by fzimmermann89 — APPROVED** [review]

**Comment by ckolbPTB** [comment]
I think any change which makes the code more general is a good idea. As this will probably require more changes than a few clean-up ones, I created an issue for this #138

**Pull request merged**